app/[locale]/(authenticated)/profile/actions.ts

```diff
@@ -0,0 +1,68 @@
+"use server";
+
+import * as z from "zod";
+import { query } from "@/app/db";
+import { Scrypt } from "oslo/password";
+import { getTranslations } from "next-intl/server";
+import { FormState } from "@/app/components/Form";
+
+const scrypt = new Scrypt();
+
+const profileValidationSchema = z
+  .object({
+    email: z.string().min(1),
+    name: z.string().min(1),
+    password: z.union([z.string().min(8), z.literal("")]),
+    confirmPassword: z.string(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    path: ["confirm_password"],
+  });
+
+export default async function updateProfile(
+  _prevState: FormState,
+  data: FormData
+): Promise<FormState> {
+  const t = await getTranslations("ProfileView");
+  const request = profileValidationSchema.safeParse(
+    {
+      email: data.get("email"),
+      name: data.get("name"),
+      password: data.get("password"),
+      confirmPassword: data.get("confirm_password"),
+    },
+    {
+      errorMap: (error) => {
+        switch (error.path[0]) {
+          case "email":
+            return { message: t("errors.email_required") };
+          case "name":
+            return { message: t("errors.name_required") };
+          case "password":
+            return { message: t("errors.password_format") };
+          case "confirm_password":
+            return { message: t("errors.password_confirmation") };
+          default:
```

Comment:
We need to show a different error message if the email isn't a valid email, or the password isn't long enough. This form should have the same validation errors as the accept invite action `(public)/invite/actions.ts` except that the password and confirm_password should not be required

Reply:
Ah, makes sense. What other messages would we display for the password here?

Reply:
We'll want to show a message if the password isn't log enough and if the confirm password doesn't match. So since the password field isn't required, I think you might just need to distinguish between the email required message and the email format message